R/busy-indicators.R

```diff
@@ -46,15 +46,24 @@ useBusyIndicators <- function(..., spinners = TRUE, pulse = TRUE) {
 
   rlang::check_dots_empty()
 
-  attrs <- list("shinyBusySpinners" = spinners, "shinyBusyPulse" = pulse)
+  js <- c()
 
-  js <- vapply(names(attrs), character(1), FUN = function(key) {
-    if (attrs[[key]]) {
-      sprintf("document.documentElement.dataset.%s = 'true';", key)
-    } else {
-      sprintf("delete document.documentElement.dataset.%s;", key)
+  if (rlang::is_logical(spinners) || rlang::is_string(spinners)) {
+    if (rlang::is_string(spinners)) {
+      spinners <- rlang::arg_match(spinners, .spinner_names)
```

Comment:
I was going to add a `spinner_type` argument to `busyIndicatorOptions()` for this...